File: Services/Movies/Movies.API/Models/CreateOrUpdateMovieRequest.cs
using System.ComponentModel.DataAnnotations;
using Movies.API.Constants;

namespace Movies.API.Models;

public class CreateOrUpdateMovieRequest
{
    public int? Id { get; set; }
    
    [Required(ErrorMessage = ErrorMessages.TITLE_REQUIRED)]
    public string Title { get; set; }
    
    [Required(ErrorMessage = ErrorMessages.LENGTH_REQUIRED)]
    public int Length { get; set; }
    
    [Required(ErrorMessage = ErrorMessages.YEAR_REQUIRED)]
    public uint Year { get; set; }
    
    public string? TrailerUrl { get; set; }
    public string? ImageUrl { get; set; }
    public string? ImdbUrl { get; set; }
    
    [Required(ErrorMessage = ErrorMessages.GENRES_REQUIRED)]
    public List<int> Genres { get; set; }
    
    [Required(ErrorMessage = ErrorMessages.PEOPLE_AND_ROLES_REQUIRED)]
    public List<PersonRole> People { get; set; }
}

File: app/src/models/Movie/CreateOrUpdateMovieRequest.ts
export default class CreateOrUpdateMovieRequest {
    constructor(
        public id: number = -1,
        public title: string = '',
        public year: number = 0,
        public length: number = 0,
        public trailerUrl: string | null = '',
        public imdbUrl: string | null = '',
        public imageUrl: string | null = '',
        public people: {
            personId: number;
            roleId: number;
        }[] = [],
        public genres: number[]
    ) {}
}
